spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/data/parquet/vectorized/TestParquetDictionaryEncodedVectorizedReads.java

```diff
@@ -33,11 +39,35 @@
 import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
 import org.apache.iceberg.relocated.com.google.common.collect.Iterables;
 import org.apache.iceberg.spark.data.RandomData;
+import org.apache.iceberg.spark.data.TestHelpers;
+import org.apache.iceberg.spark.data.vectorized.VectorizedSparkParquetReaders;
+import org.apache.iceberg.types.Types;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.vectorized.ColumnarBatch;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 
 public class TestParquetDictionaryEncodedVectorizedReads extends TestParquetVectorizedReads {
 
+  protected static SparkSession spark = null;
```

Comment:
can we test this without having to start up spark? Also it seems like `IntBackedDecimalDictEncodedReader`/ `LongBackedDecimalDictEncodedReader` / `VarWidthBinaryDictEncodedReader` / `FixedLengthDecimalDictEncodedReader` have been affected, so maybe there should be tests for these too?

Reply:
Using the Spark `DataFrameReader` to read the Parquet file is the easiest way for the test verification.

IIUC, if a decimal can be represented by 4 bytes, we use `IntBackedDecimalDictEncodedReader` and if it cannot be represented by 4 bytes but can be by 8 bytes, we use `LongBackedDecimalDictEncodedReader`, but can you please tell me what types use `VarWidthBinaryDictEncodedReader` / `FixedLengthDecimalDictEncodedReader`?

Reply:
Hmm, afaics, `VectorizedParquetDefinitionLevelReader.FixedLengthDecimalReader` is never used, and thus neither is `VectorizedDictionaryEncodedParquetValuesReader.FixedLengthDecimalDictEncodedReader`.

Reply:
Hmm, even ` VectorizedParquetDefinitionLevelReader.IntBackedDecimalReader` and `VectorizedParquetDefinitionLevelReader.LongBackedDecimalReader` are never used, afait.
In `VectorizedParquetDefinitionLevelReader`, the methods `fixedLengthDecimalReader()`, `intBackedDecimalReader()` and `longBackedDecimalReader()` are never called.

I see now that `VectorizedPageIterator.IntBackedDecimalPageReader`, `VectorizedPageIterator.LongBackedDecimalPageReader`, `VectorizedPageIterator::intBackedDecimalPageReader()`, and `VectorizedPageIterator::longBackedDecimalPageReader()` were deprecated in https://github.com/apache/iceberg/pull/3249 and subsequently all removed in Iceberg 1.4.0. So we have some dead code now.

Reply:
Shouldn't we remove the dead code then instead of fixing it?